ops/model.py

```diff
@@ -1118,8 +1122,14 @@ def restart(self, *service_names: str):
         if not service_names:
             raise TypeError('restart expected at least 1 argument, got 0')
 
-        self._pebble.stop_services(service_names)
-        self._pebble.start_services(service_names)
+        try:
+            self._pebble.restart_services(service_names)
+        except pebble.APIError as e:
+            if e.code != 400:
+                raise e
+            # support old behaviour with old pebble instances
```

Comment:
Maybe clarify, something like: `# support old Pebble instances that don't support "restart" action`